e2e_tests/tests/deploy/test_local.py

```diff
@@ -192,3 +197,79 @@ def test_agent_up_down() -> None:
     assert len(containers) == 0
 
     master_down([])
+
+
+@pytest.mark.parametrize("steps", [10])
+@pytest.mark.parametrize("num_agents", [3, 5])
+@pytest.mark.parametrize("should_disconnect", [False, True])
+@pytest.mark.det_deploy_local
+def test_stress_agents_reconnect(steps: int, num_agents: int, should_disconnect: bool) -> None:
+    master_host = "localhost"
+    master_port = "8080"
+    conf.MASTER_IP = master_host
+    conf.MASTER_PORT = master_port
+    master_up([])
+
+    # Start all agents.
+    agents_are_up = [True] * num_agents
+    for i in range(num_agents):
+        agent_up(["--agent-name", f"agent-{i}"])
+    time.sleep(3)
+
+    for _ in range(steps):
+        for agent_id, agent_is_up in enumerate(agents_are_up):
+            if random.choice([True, False]):  # Flip agents status randomly.
+                continue
+
+            if agent_is_up:
+                if should_disconnect:
+                    agent_down(["--agent-name", f"agent-{agent_id}"])
+                else:
+                    with logged_in_user(ADMIN_CREDENTIALS):
+                        subprocess.run(
+                            [
+                                "det",
+                                "-m",
+                                conf.make_master_url(),
+                                "agent",
+                                "disable",
+                                f"agent-{agent_id}",
+                            ]
+                        )
+            else:
+                if should_disconnect:
+                    agent_up(["--agent-name", f"agent-{agent_id}"])
+                else:
+                    with logged_in_user(ADMIN_CREDENTIALS):
+                        subprocess.run(
+                            [
+                                "det",
+                                "-m",
+                                conf.make_master_url(),
+                                "agent",
+                                "enable",
+                                f"agent-{agent_id}",
+                            ]
+                        )
+            agents_are_up[agent_id] = not agents_are_up[agent_id]
+        time.sleep(10)
+
+        # Validate that our master kept track of the agent reconnect spam.
+        agent_list = json.loads(
+            subprocess.check_output(
+                [
+                    "det",
+                    "agent",
+                    "list",
+                    "--json",
+                ]
+            ).decode()
+        )
+        assert sum(agents_are_up) <= len(agent_list)
+        for agent in agent_list:
+            agent_id = int(agent["id"].replace("agent-", ""))
+            assert agents_are_up[agent_id] == agent["enabled"]
+
+    for agent_id in range(num_agents):
```

Comment:
it will be useful to also run a command and/or noop experiment to make sure the master can still schedule things? there were failures when `det agent list` still works okay but nothing would schedule because the resource pool is in a bad state.

Reply:
Good call out.

The no-op experiment was originally failing with unable to connect to the logging driver at 24224.

Seems my multiple deploy agents locally caused them all to try and create the same fluent container name causing it constantly restart.